src/precompile.jl

```diff
@@ -0,0 +1,24 @@
+using PrecompileTools: @setup_workload, @compile_workload
+
+@setup_workload begin
+    # TODO: Are these all safe to call here and bake into the pkgimage?
+    Connections.__init__()
+    MultiPartParsing.__init__()
+    Parsers.__init__()
+    ConnectionRequest.__init__()
+
+    gzip_data(data::String) = read(GzipCompressorStream(IOBuffer(data)))
+
+    # random port in the dynamic/private range (49152–65535) which are are
+    # least likely to be used by well-known services
+    _port = 57813
+
+    server = HTTP.serve!("0.0.0.0", _port; verbose = -1) do req
```

Comment:
You can pass `listenany=true` to `serve!` which will ensure a non-used random port is selected. You can then call `HTTP.port(server)` to get the port that was selected.

Reply:
Great, changed.